rotkehlchen/globaldb/updates.py

```diff
@@ -300,74 +341,232 @@ def _parse_full_insert(self, insert_text: str) -> AssetData:
+            with connection.savepoint_ctx() as cursor:
+                executeall(cursor, action)
+                if local_asset is not None:
+                    AssetResolver().clean_memory_cache(identifier=local_asset.identifier)
+        except sqlite3.Error:  # https://docs.python.org/3/library/sqlite3.html#exceptions
+            if local_asset is None:
+                try:  # if asset is not known then simply do an insertion
+                    with connection.savepoint_ctx() as cursor:
+                        executeall(cursor, full_insert)
+                except sqlite3.Error as e:
+                    self.msg_aggregator.add_warning(
+                        f'Failed to add asset {remote_asset_data.identifier} in the '
+                        f'DB during the v{version} assets update. Skipping entry. '
+                        f'Error: {str(e)}',
+                    )
+                return  # fail or succeed continue to next entry
+
+            if local_asset is not None:
+                AssetResolver().clean_memory_cache(local_asset.identifier.lower())
+
+            # otherwise asset is known, so it's a conflict. Check if we can resolve
+            resolution = assets_conflicts.get(local_asset) if assets_conflicts else None
+            if resolution == 'local':
+                # do nothing, keep local
+                return
+            if resolution == 'remote':
+                try:
+                    with connection.savepoint_ctx() as cursor:
+                        _force_remote_asset(cursor, local_asset, full_insert)
+                except sqlite3.Error as e:
+                    self.msg_aggregator.add_warning(
+                        f'Failed to resolve conflict for {remote_asset_data.identifier} in '
+                        f'the DB during the v{version} assets update. Skipping entry. '
+                        f'Error: {str(e)}',
+                    )
+                return  # fail or succeed continue to next entry
+
+            # else can't resolve. Mark it for the user to resolve.
+            # TODO: When assets refactor is finished, remove the usage of AssetData here
+            local_data = GlobalDBHandler().get_all_asset_data(  # pylint: disable=unsubscriptable-object  # noqa: E501
+                mapping=False,
+                serialized=False,
+                specific_ids=[local_asset.identifier],
+            )[0]
+            self.conflicts.append((local_data, remote_asset_data))
+
     def _apply_single_version_update(
             self,
-            connection: DBConnection,
+            connection: 'DBConnection',
             version: int,
             text: str,
-            conflicts: Optional[dict[Asset, Literal['remote', 'local']]],
+            assets_conflicts: Optional[dict[Asset, Literal['remote', 'local']]],
+            update_file_type: UpdateFileType,
     ) -> None:
+        """
+        Process each action in the queried data and try the following in order:
```

Comment:
Why did you copy paste my previous docstring suggestion when you now have actually split it in 3 files? The docstring is no longer relevant and needs changing.